lib/lhm/entangler.rb

```diff
@@ -95,6 +101,40 @@ def revert
 
     private
 
+    def special_origin?
+      TABLES_WITH_LONG_QUERIES.include? @origin.name
+    end
+
+    def kill_long_running_queries_on_origin_table!
+      return unless ENV['LHM_KILL_LONG_RUNNING_QUERIES'] == 'true'
+      3.times do
+        long_running_queries(@orogin.name).each do |id, query, duration|
+          Lhm.logger.info "Action on table #{table_name} detected; killing #{duration}-second query: #{query}."
+          @connection.execute("KILL #{id}")
+        end
+        sleep(7)
+      end
+    end
+
+    def long_running_queries(table_name)
+      result = @connection.execute <<-SQL.strip_heredoc
+        SELECT ID, INFO, TIME FROM INFORMATION_SCHEMA.PROCESSLIST
+        WHERE command <> 'Sleep'
+          AND INFO LIKE '%FROM `#{table_name}`%'
+          AND INFO NOT LIKE "%INFORMATION_SCHEMA.PROCESSLIST%"
+          AND TIME > 10 ORDER BY TIME DESC
+      SQL
+      result.to_a.compact
+    end
+
+    def execute_with_timeout(stmt, sec)
+      Timeout.timeout(sec) do
```

Comment:
I am strongly opposed to using Timeout here! We could be causing all kinds of carnage inside of LHM, the MySQL driver, the DB…

See http://www.mikeperham.com/2015/05/08/timeout-rubys-most-dangerous-api/

Can we use a session-local MySQL timeout variable instead? That will ensure resources are cleaned up.